server/src/main/java/org/opensearch/search/aggregations/bucket/histogram/DateHistogramAggregator.java

```diff
@@ -201,13 +203,28 @@ public LeafBucketCollector getLeafCollector(LeafReaderContext ctx, LeafBucketCol
             return LeafBucketCollector.NO_OP_COLLECTOR;
         }
 
-        SortedNumericDocValues values = valuesSource.longValues(ctx);
+        final SortedNumericDocValues values = valuesSource.longValues(ctx);
+        final NumericDocValues singleton = DocValues.unwrapSingleton(values);
+
+        if (singleton != null) {
+            // Optimized path for single-valued fields
+            return new LeafBucketCollectorBase(sub, values) {
+                @Override
+                public void collect(int doc, long owningBucketOrd) throws IOException {
+                    if (singleton.advanceExact(doc)) {
+                        long value = singleton.longValue();
+                        collectValue(sub, doc, owningBucketOrd, value);
+                    }
```

Comment:
I am not completely sure, how the current single-valued path is optimized. That being said, I am wondering if we can reuse the `preparedRounding` and whether `hardBounds` contain the same or not, across collect invocations

Reply:
Never mind, just realized that these values are within a single document. So, they can be different across documents